packages/sanity/src/core/config/auth/types.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Authentication options
+ *
+ * @public
+ */
+export interface AuthConfig {
+  /**
+   * Login method to use for the studio the studio. Can be one of:
+   * - `dual` (default) - attempt to use cookies where possible, falling back to
+   *   storing authentication token in `localStorage` otherwise
+   * - `cookie` - explicitly disable `localStorage` method, relying only on cookies. May fail due
+   *   to cookies being treated as third-party cookies in some browsers, thus the default is `dual`.
+   * - `token` - explicitly disable cookies, relying only on `localStorage` method
+   */
+  loginMethod?: 'dual' | 'cookie' | 'token'
+
+  /**
+   * Whether to append the providers specified in `providers` with the default providers from the
+   * API, or replace the default providers with the ones specified.
+   */
+  mode?: 'append' | 'replace'
```

Comment:
Just a thought, but would it be slightly better if we typed `providers` as `Provider[] | {mode: 'append' | 'replace'; providers: Provider[]}`?

Reasoning: `mode` is a pretty broad term, and having it on the authconfig itself makes it seem like it's about "auth mode" (whatever that would be), so maybe more intuitive to make it part of the providers config key

Reply:
I agree - I think this is directly ported from v2, so the old names stick around. I'll see if I can deprecate the old way of doing things.

Reply:
After thinking about this some more, I actually liked the idea of making this a _reducer_ better, to align with other APIs. I've deprecated the `mode` property, and `providers` is now _either_ a static array (functioning as before, in combination with `mode` for backwards compatibility) or a function that returns an array/promise that resolves to an array.

Let me know if you agree with this approach!